openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -116,137 +116,59 @@ public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi, Map<String, Strin
-        }
-    }
-
-
-    private static void updateComponentHeaderSection(Components components, String schemaName, String modifiedName,
-                                                     Map<String, Header> modifiedHeaders) {
-        if (components.getHeaders() != null) {
-            Map<String, Header> headers = components.getHeaders();
-            for (Map.Entry<String, Header> header : headers.entrySet()) {
-                Header headerValue = header.getValue();
-                Schema<?> type = headerValue.getSchema();
-                String ref = type.get$ref();
-                if (ref != null) {
-                    updateRef(schemaName, modifiedName, type);
-                }
-                headerValue.setSchema(type);
-                header.setValue(headerValue);
-                modifiedHeaders.put(header.getKey(), headerValue);
-            }
-        }
-    }
-
-    private static void updateComponentParameterSection(Components components, String schemaName, String modifiedName,
-                                                        Map<String, Parameter> modifiedReusableParameters) {
-        if (components.getParameters() != null) {
-            Map<String, Parameter> parameters = components.getParameters();
-            for (Map.Entry<String, Parameter> param : parameters.entrySet()) {
-                Parameter parameter = param.getValue();
-                if (parameter.getSchema() != null) {
-                    Schema<?> paramSchema = parameter.getSchema();
-                    String ref = paramSchema.get$ref();
-                    if (ref != null) {
-                        updateRef(schemaName, modifiedName, paramSchema);
-                    }
-                    parameter.setSchema(paramSchema);
-                }
-                modifiedReusableParameters.put(param.getKey(), parameter);
-            }
-        }
-    }
-
-    private static void updateComponentResponseSection(Components components, String schemaName, String modifiedName,
-                                                       Map<String, ApiResponse> modifiedResponses) {
-        if (components.getResponses() != null) {
-            Map<String, ApiResponse> responsesEntry = components.getResponses();
-            for (Map.Entry<String, ApiResponse> response : responsesEntry.entrySet()) {
-                ApiResponse modifiedResponse = updateSchemaTypeForResponses(schemaName, modifiedName,
-                        response.getValue());
-                modifiedResponses.put(response.getKey(), modifiedResponse);
+        String openApiJson = Json.pretty(openapi);
+        for (Map.Entry<String, String> entry : nameMap.entrySet()) {
+            String schemaName = entry.getKey();
+            String modifiedName = entry.getValue();
+            if (schemaName.equals(modifiedName)) {
+                continue;
             }
-        }
-    }
+            String patternString = ("(\\\"\\$ref\\\"\\s*:\\s*\\\"#/components/schemas/%s\\\")|" +
```

Comment:
Shall we extract this String template into a constant ? Um referring to the following part only;
```
"(\\\"\\$ref\\\"\\s*:\\s*\\\"#/components/schemas/%s\\\")|('\\$ref'\\s*:\\s*'#/components/schemas/%s')"
```

Reply:
Fixed via [21628fe](https://github.com/ballerina-platform/openapi-tools/pull/1766/commits/21628fe6c77a0f54d70984c8e48d6718612b1319)